src/supportbase.c

```diff
@@ -619,15 +623,16 @@ void sbRebuildULCfg(base_game_info_t **list, const char *prefix, int gamecount,
 
         memset(&GameEntry, 0, sizeof(GameEntry));
         GameEntry.Byte08 = 0x08; // just to be compatible with original ul.cfg
-        strcpy(GameEntry.startup, "ul.");
+        strcpy(GameEntry.magic, "ul.");
 
         for (i = 0; i < gamecount; i++) {
             game = &(*list)[i];
 
             if (game->format == GAME_FORMAT_USBLD && (i != excludeID)) {
-                memset(&GameEntry.startup[3], 0, GAME_STARTUP_MAX);
+                memcpy(GameEntry.startup, game->startup, GAME_STARTUP_MAX);
+                GameEntry.startup[GAME_STARTUP_MAX] = '\0';
                 memcpy(GameEntry.name, game->name, UL_GAME_NAME_MAX);
-                strncpy(&GameEntry.startup[3], game->startup, GAME_STARTUP_MAX);
+                GameEntry.name[UL_GAME_NAME_MAX] = '\0';
```

Comment:
and this one

Reply:
bu then 32 long character names will become 31 character long - the last symbol will be cut? Keep in mind that for UL we dont align BUFFER +1, but BUFFER. Does that mean that after some game deletion all 32 character names will become cropped?

Reply:
I’m not sure about the cropping tbh, it’s a little confusing and.. stupid because base_game_info_t which OPL stores the info as is +1 but USBExtreme_game_entry_t isn’t so on those lines I commented like you said it’s BUFFER which means the array limit is BUFFER -1 since the array starts at 0

Reply:
Perhaps that is the cause of the 31 character limit bug? Once a game is renamed or deleted and the list is updated?

Reply:
USBExtreme_game_entry_t is made in such a way so it doesn't have space for leading zero for char strings. Char strings can fill the whole buffer, maybe this was the problem that @HowlingWolfHWC faced. In fact, I now understand that we should not fill it with leading zero at all, cause GameEntry.name can not contain zeroes at all (in the case of 32 character long names). This is an issue cause we cannot allocate a larger buffer - we will break USBExtreme_game_entry_t totally - all offsets will become wrong.

Reply:
Makes sense and that’s exactly what I did with adding +1 to the name lol